NGitLab.Tests/Docker/GitLabDockerContainer.cs

```diff
@@ -295,19 +298,29 @@ async Task GenerateAdminToken(GitLabCredential credentials)
             TestContext.Progress.WriteLine("Generating Credentials");
 
             var url = await GetCurrentUrl(page);
+            var gitLabVersionAsNuGetVersion = NuGetVersion.Parse(GitLabDockerVersion);
 
             // Login
             if (url == "/users/sign_in")
             {
-                await page.Locator("form#new_user input[name='user[login]']").FillAsync(AdminUserName);
-                await page.Locator("form#new_user input[name='user[password]']").FillAsync(AdminPassword);
+                if (VersionRange.Parse("[16.0,17.0)").Satisfies(gitLabVersionAsNuGetVersion))
```

Comment:
Could you make the range to be version 16 and upper (not limiting between 16 and 17), cause it would make a potential version 18 to throw (and ideally they don't change their html markup again on future major versions ^^).

Should maybe be `[16.0,)` ?

Also, as the version ranges are repeated, could we have something like:
```csharp
// Declare supported ranges
var isMajorVersion15 = VersionRange.Parse("[15.0,16.0)").Satisfies(gitLabVersionAsNuGetVersion);
var isMajorVersionAtLeast16 = VersionRange.Parse("[16.0,)").Satisfies(gitLabVersionAsNuGetVersion);

// Then, instead of having to throw at 2 places, it can be checked early
if  (!isMajorVersion15  && !isMajorVersionAtLeast16)
{
    throw new NotImplementedException($"Unable to generate an admin token: selected GitLabDockerVersion '{GitLabDockerVersion}' doesn't match any supported range.")
}

// Then variables can be reused for the conditions
if (isMajorVersionAtLeast16)
{
  // playwright code for 16+
}
else
{
  // playwright code for 15
}
```

Reply:
Hi @Toa741,

I agree with you. Unfortunately for version 17 there needs to be yet another implementation - I already tested the current code with 17.0.1 and it didn't work.

How do you suggest to proceed?

Reply:
Oh, then in this case we can keep the range you had ([16.0,17.0))

Reply:
Extraction of supported ranges: Done